src/main/java/com/couchbase/lite/support/MultipartReader.java

```diff
@@ -69,10 +68,9 @@ private boolean memcmp(byte[] array1, byte[] array2, int len) {
     }
 
     public Range searchFor(byte[] pattern, int start) {
-
         KMPMatch searcher = new KMPMatch();
-        int matchIndex = searcher.indexOf(buffer.toByteArray(), pattern, start);
-
+        int buffLen = Math.min(buffer.length(), buffer.buffer().length);
```

Comment:
`ByteArrayBuffer.buffer()` allows to access `byte[]` in `ByteArrayBuffer`.  So, `buffer().length` is size of `byte[]`. `ByteArrayBuffer. length()` returns length of actual data that is stored in `byte[]`. In general, `length() <= buffer().length`. But `length()` of ByteArrayBuffer could be longer than buffer().length because its value is changeable by calling 'setLength(int)'. The code of line 72 makes sure following code to access invalid data.